### PR TITLE
chore: theme token cleanup + chrome-active helper

### LINKED ISSUE #102: chore(theme): promote --frame-* layout tokens to Tailwind namespaces
## Background

Follow-up from [a review comment on #101](https://github.com/decdn/website/pull/101#discussion_r3231341458) (`gemini-code-assist`, medium): #101 (closing #78) migrated the **typography** tokens to Tailwind v4's `--text-*` namespace (Approach A — real `text-h1`, `text-body`, … utilities), but kept the **layout** `--frame-*` tokens as raw vars consumed via arbitrary-value classes (Approach B — `px-[var(--frame-gutter)]`, `max-w-[var(--frame-max)]`, …), matching the existing `scroll-mt-[var(--nav-h)]` precedent. This was a deliberate split (see #78's "Approach (open question)" section). This issue tracks the remaining "Approach A for layout" option, should we want it.

## Proposal

Promote the layout tokens in `app/globals.css`'s `@theme inline` block to Tailwind v4 namespaces so standard utilities auto-generate (no bracket-soup):

- `--frame-gutter: clamp(1.5rem, 4vw, 4rem)` → `--spacing-frame-gutter` ⇒ `px-frame-gutter`
- `--frame-pad-y: clamp(5rem, 9vw, 7rem)` → `--spacing-frame-pad-y` ⇒ `py-frame-pad-y`
- `--frame-max: 1280px` → `--container-frame` ⇒ `max-w-frame` (also yields a `@frame` container-query variant — harmless)
- `--frame-min-h-cap: 54rem` → fold the `min()` into the token: `--spacing-frame-min-h: min(100svh, 54rem)` ⇒ `min-h-frame-min-h` (alternatively keep `--frame-min-h-cap` as a raw var and leave `min-h-[min(100svh,var(--frame-min-h-cap))]` as-is — it's irreducibly an arbitrary value otherwise)

### Sites to update

- `components/ui/Frame.tsx` — `px-frame-gutter py-frame-pad-y min-h-frame-min-h` on the `<section>`, `max-w-frame` on the inner `<div>`
- `components/site/Footer.tsx` — `px-frame-gutter`, `max-w-frame`
- `components/site/Chrome.tsx` — `px-frame-gutter`, `max-w-frame`
- `app/globals.css` — the lone CSS ref `right: var(--frame-gutter)` (mobile-menu close button) → `var(--spacing-frame-gutter)`; update the `--frame-*` mentions in the `:root` `--nav-h` comment and the `@theme inline` comments

## Trade-off

Renames the public CSS-var API (`--frame-*` → `--spacing-frame-*` / `--container-frame`). No external consumers were found when #78 was scoped (`docs/`, `content/blog/`, all `.css`/`.ts(x)` — `app/globals.css` is the website's only stylesheet), so this is low-risk; bundle as one PR (a partial migration reintroduces the inconsistency #78 fixed).

## Verification

Same triple as #78: `pnpm lint && pnpm exec tsc --noEmit && pnpm test && pnpm build`; `pnpm dev` smoke (frame gutter / max-width rail / min-height unchanged across §01–§05, `/blog`, a post); confirm `out/_next/static/chunks/*.css` is roughly flat in size.

## Links

- Source comment: https://github.com/decdn/website/pull/101#discussion_r3231341458
- PR that landed the typography half + Approach-B layout: #101
- Original migration issue (with the A/B discussion): #78

### LINKED ISSUE #105: chore(theme): decide on the unreferenced --font-sans token
## Background

`app/globals.css`'s `@theme inline` block defines `--font-sans: var(--font-geist-sans)`, but nothing in `app/`, `components/`, `lib/`, or `content/` uses the `font-sans` utility class or `var(--font-sans)`. Surfaced while auditing the `@theme inline` block for #103 — unlike `--text-display` (a truly dead type-scale rung), `--font-sans` isn't dead code in the same way: in Tailwind v4 it *overrides* the built-in `font-sans` utility's `font-family`. So it's a judgment call, not an obvious deletion.

Complicating it: the global typeface is already applied directly in the `html, body` rule via `font-family: var(--font-geist-sans), ui-sans-serif, system-ui, …` — that rule does **not** route through `var(--font-sans)`. So today the `--font-sans` token only matters if someone writes `class="font-sans"` somewhere (nobody does), and even then it wouldn't pick up the `ui-sans-serif, system-ui, …` fallback stack the `html, body` rule spells out.

## Options

1. **Delete it** — one-line removal from `@theme inline`. The `font-sans` utility falls back to Tailwind's default sans stack; the page's actual typeface is unaffected because `html, body` sets `font-family` directly. Simplest, matches the #103 cleanup.
2. **Keep it, and make it the source of truth** — point the `html, body` `font-family` at `var(--font-sans)` (including the `ui-sans-serif, system-ui, …` fallbacks in the token value), so there's one place that defines "our sans". More cohesive, slightly more churn.
3. **Keep as-is** — leave the token; it's harmless and means a future `font-sans` class would get the Geist face.

No strong recommendation — (1) if we like the #103-style "remove what nothing references" rule, (2) if we'd rather consolidate the font definition.

## Links

- Surfaced in #103 (kept out of scope there — different category from the dead `--text-display` rung).
- Related theme follow-ups: #102, #103.

### LINKED ISSUE #117: chrome: extract effectiveActive into pure helper + add unit test
Tracking suggestion from the [PR #114 review](https://github.com/decdn/website/pull/114).

## Background

After #114, `components/site/Chrome.tsx` derives the navbar tone from a single ternary at render:

```ts
const effectiveActive = pathname === "/" && scrolled ? active : "intro";
```

This is the load-bearing fix for #112 (and the back-nav flash). It is currently uncovered by tests — and the path of least resistance to cover it doesn't require pulling in `jsdom` / RTL / `IntersectionObserver` mocks.

## Proposal

Extract the ternary into a pure helper, then unit-test the helper.

```ts
// components/site/chrome-active.ts
export function resolveActiveSection<T extends string>(
  pathname: string, scrolled: boolean, observed: T, fallback: T,
): T {
  return pathname === "/" && scrolled ? observed : fallback;
}
```

`Chrome.tsx` then calls `resolveActiveSection(pathname, scrolled, active, "intro")`.

## Why a separate issue

- The repo currently has no component tests; `vitest.config.ts` only includes `**/*.test.ts` (note: not `.tsx`) and there is no `jsdom` / `@testing-library/*` setup. A pure-function helper sidesteps that entirely.
- Out of scope for the original bug fix in #114 — tracked here so the coverage gap is documented.

## Test cases (sketch)

The helper test (`components/site/chrome-active.test.ts`) should cover:

1. `pathname="/blog/foo/", scrolled=true, observed="compare"` → `"intro"` (the actual #112 bug).
2. `pathname="/", scrolled=false, observed="compare"` → `"intro"` (post-nav, pre-scroll — the back-nav flash).
3. `pathname="/", scrolled=true, observed="compare"` → `"compare"` (passthrough).
4. `pathname="/", scrolled=true, observed="faq"` → `"faq"` (other `DARK_SECTIONS` member, guards against accidental constant return).
5. `pathname="/blog/", scrolled=false, observed="faq"` → `"intro"` (off-home dominates `scrolled`).

## Optional follow-on

Widen `vitest.config.ts` `include` to `**/*.test.{ts,tsx}` so future component-adjacent helpers can co-locate.

### PR DESCRIPTION
**Pull request by yigitdot**
## Summary

- Drop the unreferenced `--font-sans` token from `@theme inline` — `html, body` already applies the typeface directly and no component uses the `font-sans` utility (closes #105).
- Promote the remaining `--frame-*` layout tokens to Tailwind v4 namespaces (`--spacing-frame-*` / `--container-frame`), so `px-frame-gutter` / `py-frame-pad-y` / `max-w-frame` / `min-h-frame-min-h` utilities replace the `[var(--frame-*)]` arbitrary-value classes left over from #101. Finishes the Approach-A migration started in #78 (closes #102).
- Extract the home-route-gated nav-tone ternary in `Chrome.tsx` into a pure `resolveActiveSection` helper (rationale doc co-located with the function) and add 6 unit tests covering both regression guards, all four AND-gate quadrants, and the exact-match boundary — no jsdom needed (closes #117).

#115 was reviewed in parallel and intentionally deferred (audit comment posted on the issue); the helper extracted here is where its routing-aware fix will live.

## Test plan

- [x] `pnpm lint` — clean
- [x] `pnpm exec tsc --noEmit` — clean
- [x] `pnpm test` — 101 pass, including the 6 `resolveActiveSection` cases
- [x] `pnpm build` — static export clean; built CSS contains `.px-frame-gutter` / `.py-frame-pad-y` / `.max-w-frame` / `.min-h-frame-min-h` and zero `var(--frame-*)` refs
- [ ] `pnpm dev` smoke: gutter / max-width rail / section min-height visually unchanged on `/`, `/blog/`, and a post
- [ ] Nav tone: scroll past intro on `/` → `text-paper`; back-nav `/blog/` → `/` paints `text-ink` until first scroll; `/blog/foo/` stays `text-ink` regardless of scroll
- [ ] Mobile drawer (<768px): close `×` sits at the same gutter offset

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/globals.css [hunk 1]
```diff
@@ -6,33 +6,32 @@
...
   --color-paper: var(--paper);
   --color-whisper: var(--whisper);
-  --font-sans: var(--font-geist-sans);
 
-  /* Shared content rail: every Frame centers within --frame-max. */
-  --frame-max: 1280px;
-  --frame-gutter: clamp(1.5rem, 4vw, 4rem);
-  --frame-pad-y: clamp(
+  /* Shared content rail: every Frame centers within --container-frame. */
+  --container-frame: 1280px;
```
[the remark this answers is not in the document]
Comment (yigitdot) [reply app/globals.css:27]: Verified — `--container-*` does generate `max-w-*` in Tailwind v4. After `pnpm build` on this branch (`39ac7c8`), the static export contains `.max-w-frame{max-width:1280px}` — exactly the utility `Chrome.tsx` / `Footer.tsx` / `Frame.tsx` rely on. Tailwind v4's own defaults follow the same pattern: `node_modules/tailwindcss/theme.css` defines `--container-3xs` … `--container-7xl`, which back the framework's `max-w-3xs` … `max-w-7xl` utilities, and the official v4 docs implement every `max-w-*` as `max-width: var(--container-<name>)` (e.g. `max-w-2xs` is `max-width: var(--container-2xs)`).

The `@frame:` container-query variant the comment alludes to is an additional output of `--container-*`, not a substitute — both happen. `--width-*` isn't a max-width-generating namespace in v4, so renaming to `--width-frame` would actually break `max-w-frame`. No change needed.

**Review by alpergundogdu — APPROVED** [review]
